python_modules/dagster/dagster/_config/config_type.py

```diff
@@ -321,6 +327,13 @@ def post_process(self, value: typing.Any) -> typing.Any:
             ).format(config_value=value)
         )
 
+    def translate_enum_value(self, value):
+        """Pass through enum value as is (treats config processing as idempotent)."""
+        for ev in self.enum_values:
+            if ev.python_value == value:
+                return ev.python_value
+        return self.post_process(value)
```

Comment:
Wait I'm a bit confused. Is this even right?

Should we not be converting back to `ev.config_value`?

Reply:
Just for historical sake, this is not right - result of a mismatched understanding due to the enum bug nick discusses in a later comment.